### PR TITLE
qq: init at 2.0.3-543

### PR DESCRIPTION
**Pull request by fee1-dead**
###### Description of changes

Added https://aur.archlinux.org/packages/linuxqq-new.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/qq/default.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+, xorg
+, systemd
+, stdenv
+, at-spi2-core
+, autoPatchelfHook
+, wrapGAppsHook
+}:
+
+let
+  mirror = "https://dldir1.qq.com/qqfile/qq/QQNT/50eed662";
```
Comment (viraptor) [inline pkgs/applications/networking/instant-messengers/qq/default.nix:24]: This is used only once. Probably better to include it in `url` and make the structure simpler.

Comment (fee1-dead) [reply pkgs/applications/networking/instant-messengers/qq/default.nix:24]: will be used more if we add arm64 support

**Comment by wegank** [comment]
Having read the comments in the link, I got the impression that this version of QQ is not available to all users. Is this still the case?

Also, I would prefer that aarch64-linux support be added to the package.

**Comment by fee1-dead** [comment]
> I got the impression that this version of QQ is not available to all users. Is this still the case?

Login is still restricted to beta testers, but should be opening up soon as [a page for it](https://im.qq.com/linuxqq/index.shtml) already went public.

File: pkgs/applications/networking/instant-messengers/qq/default.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+in stdenv.mkDerivation rec {
+
+  pname = "qq";
+  version = "2.0.3-543";
+
+  src = fetchurl {
+    url = "${mirror}/QQ-v${version}_x64.deb";
```
Comment (AndersonTorres) [inline pkgs/applications/networking/instant-messengers/qq/default.nix:33]: why???

File: pkgs/applications/networking/instant-messengers/qq/default.nix [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    sha256 = "sha256-O8zaVHt/oXserPVHe/r6pAFpWFeLDVsiaazgaX7kxu8=";
+  };
+
+  unpackCmd = "mkdir root; ${dpkg}/bin/dpkg-deb -x $curSrc root";
```
Comment (AndersonTorres) [inline pkgs/applications/networking/instant-messengers/qq/default.nix:37]: have you ever read about multiline strings?

File: pkgs/applications/networking/instant-messengers/qq/default.nix (hunk 2 above)
Comment (fee1-dead) [reply pkgs/applications/networking/instant-messengers/qq/default.nix:33]: Huh? Should I use the AppImage or the RPM instead? It is proprietary

File: pkgs/applications/networking/instant-messengers/qq/default.nix (hunk 3 above)
Comment (fee1-dead) [reply pkgs/applications/networking/instant-messengers/qq/default.nix:37]: to be fair, the command isn't that long.

File: pkgs/applications/networking/instant-messengers/qq/default.nix [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    wrapGAppsHook
+    copyDesktopItems
+  ];
```
Comment (wegank) [inline pkgs/applications/networking/instant-messengers/qq/default.nix:43]:
```suggestion
let
  version = "2.0.3-543";
  srcs = {
    x86_64-linux = fetchurl {
      url = "https://dldir1.qq.com/qqfile/qq/QQNT/50eed662/QQ-v${version}_x64.deb";
      sha256 = "sha256-O8zaVHt/oXserPVHe/r6pAFpWFeLDVsiaazgaX7kxu8=";
    };
    aarch64-linux = fetchurl {
      url = "https://dldir1.qq.com/qqfile/qq/QQNT/50eed662/QQ-v${version}_arm64.deb";
      sha256 = "sha256-01ZpcoSDc5b0MCKAMq16N4cXzbouHNckOGsv+Z4et7w=";
    };
  };
  src = srcs.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
in
stdenv.mkDerivation rec {
  pname = "qq";
  inherit version src;

  unpackCmd = "dpkg-deb -x $curSrc source";

  nativeBuildInputs = [
    autoPatchelfHook
    wrapGAppsHook
    copyDesktopItems
    dpkg
  ];
```

Comment (fee1-dead) [reply pkgs/applications/networking/instant-messengers/qq/default.nix]: Done. Thanks for the suggestion

File: pkgs/applications/networking/instant-messengers/qq/default.nix [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      icon = "qq";
+      exec = "qq";
+      name = "qq";
+    })
+  ];
+
+  meta = with lib; {
+    homepage = "https://im.qq.com/linuxqq/";
+    description = "Messaging app";
+    platforms = [ "x86_64-linux" ];
```
Comment (wegank) [inline pkgs/applications/networking/instant-messengers/qq/default.nix:95]: Oops. Also this.

```suggestion
    platforms = [ "x86_64-linux" "aarch64-linux" ];
```

**Review by wegank — APPROVED** [review]
LGTM + Tested on aarch64-linux.

File: pkgs/applications/networking/instant-messengers/qq/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/applications/networking/instant-messengers/qq/default.nix]: No, I said what this mirror string is doing here. This is not as if it is a changing target.

File: pkgs/applications/networking/instant-messengers/qq/default.nix [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      sha256 = "sha256-O8zaVHt/oXserPVHe/r6pAFpWFeLDVsiaazgaX7kxu8=";
+    };
+    aarch64-linux = fetchurl {
+      url = "https://dldir1.qq.com/qqfile/qq/QQNT/50eed662/QQ-v${version}_arm64.deb";
+      sha256 = "sha256-01ZpcoSDc5b0MCKAMq16N4cXzbouHNckOGsv+Z4et7w=";
+    };
+  };
+  src = srcs.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+in
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/applications/networking/instant-messengers/qq/default.nix:39]: I have a feel this rec is not needed. After all, the variables used below were defined in the let above.
```suggestion
stdenv.mkDerivation {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Pull request merged**